Classes/Util/PBTerminalUtil.m

```diff
@@ -7,23 +7,36 @@
 
 #import "PBTerminalUtil.h"
 #import "Terminal.h"
+#import "iTerm2GeneratedScriptingBridge.h"
 
 @implementation PBTerminalUtil
 
-+ (NSString *) initialCommand:(NSURL *)workingDirectory {
-	return [NSString stringWithFormat:@"cd \"%@\"; clear; echo '# Opened by GitX'; ",
-			workingDirectory.path];
++ (NSString *) terminalCommand:(NSString *)command inDirectory:(NSURL *)directory {
+	return [NSString stringWithFormat:@"cd \"%@\"; clear; echo '# Opened by GitX'; %@",
+			directory.path, command];
 }
 
++ (NSString *) iTerm2Command:(NSString *)command inDirectory:(NSURL *)directory {
+	return [NSString stringWithFormat:@"/usr/bin/login -f %@ /bin/sh -c 'cd \"%@\"; clear; echo \"# Opened by GitX\"; %@; ${SHELL} -l",
+			NSUserName(), directory.path, command];
+}
 + (void) runCommand:(NSString *)command inDirectory:(NSURL *)directory {
-	NSString * initialCommand = [self initialCommand:directory];
-	NSString * fullCommand = [initialCommand stringByAppendingString:command];
+	NSLog(@"Running command: %@ in directory %@", command, directory.path);
 	
-	TerminalApplication *term = [SBApplication applicationWithBundleIdentifier: @"com.apple.Terminal"];
-	[term doScript:fullCommand in: nil];
-	dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.1 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
-		[term activate];
-	});
+	// Prefer iTerm2. If they have it installed they probably want to use that.
+	iTerm2Application *iTerm2 = [SBApplication applicationWithBundleIdentifier: @"com.googlecode.iterm2"];
```

Comment:
You could stash that bundle ID in the preferences, so it can be selected.

Reply:
I don't understand - why would you want to change it?